### ISSUE TITLE
EOFError with ProcessPool

### ISSUE THREAD

**Issue by MrAlexBailey** [noxdafox/pebble#35]
```
from pebble import ProcessPool
from concurrent.futures import TimeoutError
import time

if __name__ == '__main__':
    with ProcessPool() as pool:
        future = pool.map(time.sleep, [1, 25, 62, 7], timeout=8)
        iterator = future.result()

        while True:
            try:
                next(iterator)
            except StopIteration:
                break
            except TimeoutError as error:
                print 'task took too long, {}, {}'.format(error.args[1], error)

        # time.sleep(2)  # <-- Fixes the crash
```

Yields:
```
task took too long, 8, ('Task timeout', 8)
task took too long, 8, ('Task timeout', 8)
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "C:\Python27\lib\multiprocessing\forking.py", line 381, in main
    self = load(from_parent)
  File "C:\Python27\lib\pickle.py", line 1384, in load
    return Unpickler(file).load()
  File "C:\Python27\lib\pickle.py", line 864, in load
    dispatch[key](self)
  File "C:\Python27\lib\pickle.py", line 886, in load_eof
    raise EOFError
EOFError
```

Env:
Windows 7
Python: 2.7.15
Pebble: 4.3.9

You can see the commented `time.sleep(2)` at the end.  Interestingly, this resolves this issue.  I suspect maybe something is happening too quickly before all the processes are closed.

I have a Ubuntu 18.04.1 LTS machine completely up to date that does not have the same issue.  So it seems windows related.

**Comment by noxdafox** [comment]
Hello and thanks for reporting this issue. 

This and issue #34 should have been fixed in the latest release `4.3.10`. If you have some spare time, could you please confirm so?

**Comment by MrAlexBailey** [comment]
Checked today and `4.3.10` is working great!  Same sample test as above.

Thank you.

**Comment by noxdafox** [comment]
No, thanks for reporting this.

**Comment by noxdafox** [comment]
FYI, I had to revert the fix as it was causing deadlock issues. I will try to find another way to resolve this.

**Comment by noxdafox** [comment]
Fix added in [release 4.6.0](https://github.com/noxdafox/pebble/releases/tag/4.6.0).

Closing this issue, please re-open it if the problem persists.
